src/search/heuristics/additive_heuristic.cc

```diff
@@ -129,7 +128,8 @@ int AdditiveHeuristic::compute_add_and_ff(const State &state) {
     return total_cost;
 }
 
-int AdditiveHeuristic::compute_heuristic(const State &state) {
+int AdditiveHeuristic::compute_heuristic(const State &ancestor_state) {
+    State state = convert_ancestor_state(ancestor_state);
```

Comment:
If I'm not overlooking anything, we could also let `Heuristic::compute_result()` convert the state. This would mean less boilerplate code and it would be simpler to implement `compute_heuristic()`, especially for newcomers.

Reply:
Ah, I guess we need this for lmcount...

Reply:
Yes, path-dependent heuristics like lmcount need to know the state from the search task. We might be able to specialize this for path-independent heuristics but it is probably not worth the effort: if we fix the component interaction problem, this conversion will disappear anyway.

Reply:
> if we fix the component interaction problem

This is becoming more and more like "If P != NP" or "If the Riemann hypothesis is true".